docs/release/release-cycle.md

```diff
@@ -0,0 +1,52 @@
+# Release Cycle
+
+The release cycle is the time between when we start working on a release on the `main` branch until the `.0` release (e.g. `v1.3.0`) is released.
+
+**Note**: For the `v1.3` and `v1.4` releases we assume each release cycle will last approximately 4 months (~ 17 weeks).
+The release cycle duration will be revisited after the `v1.4.0` release.
+
+A release cycle can be split up into the following phases:
+
+* Week 1-12: Active development
+    * All changes impacting providers' adoption of Cluster API should be implemented and merged in this period. Exceptions
+      require special approval as described in the section below.
+    * Alpha releases will be released based on the main branch only if necessary (to be determined by the release lead)
+    * Please note that we also publish daily releases based on the main branch.
+* Week 13-14: Beta
```

Comment:
Could we have a point here for creating milestone for next release. So we can mark things that won't be in this release but we will want in the next.

Reply:
Good idea. I'll split up the `Create a new GitHub milestone` task (the milestone applier stuff can only be done after the release branch has been created)

Reply:
(I'll ad this to the issue template and tasks doc, the release cycle doc currently doesn't go into this level of detail)

Reply:
Move `Create a new GitHub milestone` to week 13 and renamed to `Create a new GitHub milestone for the next release`

`Update the milestone applier config` is now part of creating a release branch which makes more sense (as soon as we create the release branch we have to change the milestone applier config so milestones are applied correctly